src/ScottPlot/Statistics/Interpolation/SplineInterpolator.cs

```diff
@@ -13,13 +13,13 @@ public abstract class SplineInterpolator
         public double[] givenYs, givenXs;
         public double[] interpolatedXs, interpolatedYs;
 
-        protected Matrix m;
-        protected MatrixSolver gauss;
+        protected Matrix? m;
+        protected MatrixSolver? gauss;
 
         protected readonly int n;
-        protected double[] a, b, c, d, h;
+        protected double[] a = null!, b = null!, c = null!, d = null!, h = null!; // they're assigned in derived classes
 
-        public SplineInterpolator(double[] xs, double[] ys, int resolution = 10)
+        protected SplineInterpolator(double[] xs, double[] ys, int resolution = 10)
```

Comment:
Might be breaking change. It's was public constructor in abstract class. Only derived classes could call it.
Binary breaking change anyway